client/components/Coverages/CoverageIcons.tsx

```diff
@@ -0,0 +1,260 @@
+    Spacer,
+} from 'superdesk-ui-framework/react';
+import {IPropsAvatarPlaceholder} from 'superdesk-ui-framework/react/components/avatar/avatar-placeholder';
+import {IPropsAvatar} from 'superdesk-ui-framework/react/components/avatar/avatar';
+import {trimStartExact} from 'superdesk-core/scripts/core/helpers/utils';
+import {getItemWorkflowStateLabel, planningUtils} from '../../utils';
+import {getVocabularyItemFieldTranslated} from '../../utils/vocabularies';
+import {getUserInterfaceLanguageFromCV} from '../../utils/users';
+import './coverage-icons.scss';
+
+interface IProps {
+    coverages: Array<DeepPartial<IPlanningCoverageItem>>;
+    users: Array<IUser>;
+    desks: Array<IDesk>;
+    contentTypes: Array<IG2ContentType>;
+    contacts?: Dictionary<IContactItem['_id'], IContactItem>;
+    tooltipDirection?: 'top' | 'right' | 'bottom' | 'left'; // defaults to 'right'
+    iconWrapper?(children: React.ReactNode): React.ReactNode;
+}
+
+const appConfig = config.appConfig as IPlanningConfig;
+
+export function isAvatarPlaceholder(
+    item: Omit<IPropsAvatar, 'size'> | Omit<IPropsAvatarPlaceholder, 'size'>
+): item is Omit<IPropsAvatarPlaceholder, 'size'> {
+    return (item as any)['kind'] != null;
+}
+
+export function getAvatarForCoverage(
+    coverage: DeepPartial<IPlanningCoverageItem>,
+    users: Array<IUser>,
+    contentTypes: Array<IG2ContentType>,
+    noIcon: boolean = false,
+): Omit<IPropsAvatar, 'size'> | Omit<IPropsAvatarPlaceholder, 'size'> {
+    const user = users.find((u) => u._id === coverage.assigned_to?.user);
+
+    const icon: {name: string; color: string} | undefined =
+        noIcon === true || coverage.planning?.g2_content_type == null ? undefined : {
+            name: trimStartExact(
+                planningUtils.getCoverageIcon(
+                    planningUtils.getCoverageContentType(
+                        coverage,
+                        contentTypes,
+                    ) || coverage.planning?.g2_content_type,
+                    coverage,
+                ),
+                'icon-',
+            ),
+            color: trimStartExact(
+                planningUtils.getCoverageIconColor(coverage) ?? '',
+                'icon--'
+            ),
+        };
+
+    if (user == null) {
+        const placeholder: Omit<IPropsAvatarPlaceholder, 'size'> = {
+            kind: 'plus-button',
+            icon: icon,
+        };
+
+        return placeholder;
+    } else {
+        const avatar: Omit<IPropsAvatar, 'size'> = {
+            initials: getUserInitials(user.display_name),
+            imageUrl: user.picture_url,
+            tooltip: user.display_name?.length > 0 ? user.display_name : null,
+            icon: icon,
+        };
+
+        return avatar;
+    }
+}
+
+export class CoverageIcons extends React.PureComponent<IProps> {
+    render() {
+        const {coverages, users} = this.props;
+
+        return (
+            <WithPopover
+                placement="bottom-end"
```

Comment:
Will this still work if the list item as at the bottom of the Browser window?

Reply:
yep, it will switch to `top-end` if there's not enough space at the bottom